File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-riskdef-5-4/PricingParameterShift.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PricingParameterShift : ISerialized
    {
        public PricingParameterShift(XmlNode xmlNode)
        {
            XmlNode parameterReferenceNode = xmlNode.SelectSingleNode("parameterReference");
            
            if (parameterReferenceNode != null)
            {
                if (parameterReferenceNode.Attributes["href"] != null || parameterReferenceNode.Attributes["id"] != null) 
                {
                    if (parameterReferenceNode.Attributes["id"] != null) 
                    {
                        parameterReferenceIDRef_ = parameterReferenceNode.Attributes["id"].Value;
                        AssetOrTermPointOrPricingStructureReference ob = new AssetOrTermPointOrPricingStructureReference(parameterReferenceNode);
                        IDManager.SetID(parameterReferenceIDRef_, ob);
                    }
                    else if (parameterReferenceNode.Attributes["href"] != null)
                    {
                        parameterReferenceIDRef_ = parameterReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        parameterReference_ = new AssetOrTermPointOrPricingStructureReference(parameterReferenceNode);
                    }
                }
                else
                {
                    parameterReference_ = new AssetOrTermPointOrPricingStructureReference(parameterReferenceNode);
                }
            }
            
        
            XmlNode shiftNode = xmlNode.SelectSingleNode("shift");
            
            if (shiftNode != null)
            {
                if (shiftNode.Attributes["href"] != null || shiftNode.Attributes["id"] != null) 
                {
                    if (shiftNode.Attributes["id"] != null) 
                    {
                        shiftIDRef_ = shiftNode.Attributes["id"].Value;
                        XsdTypeDecimal ob = new XsdTypeDecimal(shiftNode);
                        IDManager.SetID(shiftIDRef_, ob);
                    }
                    else if (shiftNode.Attributes["href"] != null)
                    {
                        shiftIDRef_ = shiftNode.Attributes["href"].Value;
                    }
                    else
                    {
                        shift_ = new XsdTypeDecimal(shiftNode);
                    }
                }
                else
                {
                    shift_ = new XsdTypeDecimal(shiftNode);
                }
            }
            
        
            XmlNode shiftUnitsNode = xmlNode.SelectSingleNode("shiftUnits");
            
            if (shiftUnitsNode != null)
            {
                if (shiftUnitsNode.Attributes["href"] != null || shiftUnitsNode.Attributes["id"] != null) 
                {
                    if (shiftUnitsNode.Attributes["id"] != null) 
                    {
                        shiftUnitsIDRef_ = shiftUnitsNode.Attributes["id"].Value;
                        PriceQuoteUnits ob = new PriceQuoteUnits(shiftUnitsNode);
                        IDManager.SetID(shiftUnitsIDRef_, ob);
                    }
                    else if (shiftUnitsNode.Attributes["href"] != null)
                    {
                        shiftUnitsIDRef_ = shiftUnitsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        shiftUnits_ = new PriceQuoteUnits(shiftUnitsNode);
                    }
                }
                else
                {
                    shiftUnits_ = new PriceQuoteUnits(shiftUnitsNode);
                }
            }
            
        
        }
        
    
        #region ParameterReference_
        private AssetOrTermPointOrPricingStructureReference parameterReference_;
        public AssetOrTermPointOrPricingStructureReference ParameterReference_
        {
            get
            {
                if (this.parameterReference_ != null)
                {
                    return this.parameterReference_; 
                }
                else if (this.parameterReferenceIDRef_ != null)
                {
                    parameterReference_ = IDManager.getID(parameterReferenceIDRef_) as AssetOrTermPointOrPricingStructureReference;
                    return this.parameterReference_; 
                }
                else
                {
                      return this.parameterReference_; 
                }
            }
            set
            {
                if (this.parameterReference_ != value)
                {
                    this.parameterReference_ = value;
                }
            }
        }
        #endregion
        
        public string parameterReferenceIDRef_ { get; set; }
        #region Shift_
        private XsdTypeDecimal shift_;
        public XsdTypeDecimal Shift_
        {
            get
            {
                if (this.shift_ != null)
                {
                    return this.shift_; 
                }
                else if (this.shiftIDRef_ != null)
                {
                    shift_ = IDManager.getID(shiftIDRef_) as XsdTypeDecimal;
                    return this.shift_; 
                }
                else
                {
                      return this.shift_; 
                }
            }
            set
            {
                if (this.shift_ != value)
                {
                    this.shift_ = value;
                }
            }
        }
        #endregion
        
        public string shiftIDRef_ { get; set; }
        #region ShiftUnits_
        private PriceQuoteUnits shiftUnits_;
        public PriceQuoteUnits ShiftUnits_
        {
            get
            {
                if (this.shiftUnits_ != null)
                {
                    return this.shiftUnits_; 
                }
                else if (this.shiftUnitsIDRef_ != null)
                {
                    shiftUnits_ = IDManager.getID(shiftUnitsIDRef_) as PriceQuoteUnits;
                    return this.shiftUnits_; 
                }
                else
                {
                      return this.shiftUnits_; 
                }
            }
            set
            {
                if (this.shiftUnits_ != value)
                {
                    this.shiftUnits_ = value;
                }
            }
        }
        #endregion
        
        public string shiftUnitsIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-riskdef-5-4/PricingParameterShift.cpp
// PricingParameterShift.cpp 
#include "PricingParameterShift.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PricingParameterShift::PricingParameterShift(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //parameterReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* parameterReferenceNode = xmlNode->FirstChildElement("parameterReference");

   if(parameterReferenceNode){parameterReferenceIsNull_ = false;}
   else{parameterReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- parameterReferenceNode , address : " << parameterReferenceNode << std::endl;
   #endif
   if(parameterReferenceNode)
   {
      if(parameterReferenceNode->Attribute("href") || parameterReferenceNode->Attribute("id"))
      {
          if(parameterReferenceNode->Attribute("id"))
          {
             parameterReferenceIDRef_ = parameterReferenceNode->Attribute("id");
             parameterReference_ = boost::shared_ptr<AssetOrTermPointOrPricingStructureReference>(new AssetOrTermPointOrPricingStructureReference(parameterReferenceNode));
             parameterReference_->setID(parameterReferenceIDRef_);
             IDManager::instance().setID(parameterReferenceIDRef_,parameterReference_);
          }
          else if(parameterReferenceNode->Attribute("href")) { parameterReferenceIDRef_ = parameterReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { parameterReference_ = boost::shared_ptr<AssetOrTermPointOrPricingStructureReference>(new AssetOrTermPointOrPricingStructureReference(parameterReferenceNode));}
   }

   //shiftNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* shiftNode = xmlNode->FirstChildElement("shift");

   if(shiftNode){shiftIsNull_ = false;}
   else{shiftIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- shiftNode , address : " << shiftNode << std::endl;
   #endif
   if(shiftNode)
   {
      if(shiftNode->Attribute("href") || shiftNode->Attribute("id"))
      {
          if(shiftNode->Attribute("id"))
          {
             shiftIDRef_ = shiftNode->Attribute("id");
             shift_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(shiftNode));
             shift_->setID(shiftIDRef_);
             IDManager::instance().setID(shiftIDRef_,shift_);
          }
          else if(shiftNode->Attribute("href")) { shiftIDRef_ = shiftNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { shift_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(shiftNode));}
   }

   //shiftUnitsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* shiftUnitsNode = xmlNode->FirstChildElement("shiftUnits");

   if(shiftUnitsNode){shiftUnitsIsNull_ = false;}
   else{shiftUnitsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- shiftUnitsNode , address : " << shiftUnitsNode << std::endl;
   #endif
   if(shiftUnitsNode)
   {
      if(shiftUnitsNode->Attribute("href") || shiftUnitsNode->Attribute("id"))
      {
          if(shiftUnitsNode->Attribute("id"))
          {
             shiftUnitsIDRef_ = shiftUnitsNode->Attribute("id");
             shiftUnits_ = boost::shared_ptr<PriceQuoteUnits>(new PriceQuoteUnits(shiftUnitsNode));
             shiftUnits_->setID(shiftUnitsIDRef_);
             IDManager::instance().setID(shiftUnitsIDRef_,shiftUnits_);
          }
          else if(shiftUnitsNode->Attribute("href")) { shiftUnitsIDRef_ = shiftUnitsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { shiftUnits_ = boost::shared_ptr<PriceQuoteUnits>(new PriceQuoteUnits(shiftUnitsNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<AssetOrTermPointOrPricingStructureReference> PricingParameterShift::getParameterReference()
{
   if(!this->parameterReferenceIsNull_){
        if(parameterReferenceIDRef_ != ""){
             return boost::shared_static_cast<AssetOrTermPointOrPricingStructureReference>(IDManager::instance().getID(parameterReferenceIDRef_));
        }else{
             return this->parameterReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AssetOrTermPointOrPricingStructureReference>();
   }
}
boost::shared_ptr<XsdTypeDecimal> PricingParameterShift::getShift()
{
   if(!this->shiftIsNull_){
        if(shiftIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDecimal>(IDManager::instance().getID(shiftIDRef_));
        }else{
             return this->shift_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDecimal>();
   }
}
boost::shared_ptr<PriceQuoteUnits> PricingParameterShift::getShiftUnits()
{
   if(!this->shiftUnitsIsNull_){
        if(shiftUnitsIDRef_ != ""){
             return boost::shared_static_cast<PriceQuoteUnits>(IDManager::instance().getID(shiftUnitsIDRef_));
        }else{
             return this->shiftUnits_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PriceQuoteUnits>();
   }
}
}

